src/default-implementations.lisp

```diff
@@ -268,6 +268,51 @@ WITH-LOCK-HELD etc etc"
   (declare (ignore condition-variable))
   (values))
 
+;;; Resource contention: semaphores
+
+(defdfun make-semaphore (&key name (count 0))
+  "Create a semaphore with the supplied NAME with initial counter value COUNT."
+  (list :lock (make-lock name)
```

Comment:
Please turn this into a struct, and add a type predicate as with the rest of the API.

Reply:
How do you want me to do that? We generally don't define structures in `default-implementations.lisp`. We do something like that for lispworks in `impl-lispworks-condition-variables`, and for digitool in `condition-variables`, but it is implementation-specific.

In `default-implementation` we cons a list in `make-lock` and return NIL for other structures.

We do `define-condition` for timeout on platforms other than SBCL in `bordeaux-threads` file. Should I do a similar thing in this file conditionalized with implemented applications so far?

```diff
diff --git a/src/bordeaux-threads.lisp b/src/bordeaux-threads.lisp
index 9749944..56fff95 100644
--- a/src/bordeaux-threads.lisp
+++ b/src/bordeaux-threads.lisp
@@ -43,6 +43,11 @@ Distributed under the MIT license (see LICENSE file)
                          (timeout-length c))
                  (format s "A timeout occurred.")))))
 
+#-(or ccl ecl sbcl)
+(defstruct %semaphore
+  lock
+  cv
+  counter)
 
 ;;; Thread Creation
 
```

below define-condition? above it? somewhere else?